### PR TITLE
Raptor adjustments

### LINKED ISSUE #43340: Pupating Zombie dying causes raptors to spawn on top of character
# Describe the bug
A pupating zombie dies roughly a maptile away, but the flesh raptors it spawns will appear around the player.

# Expected behavior
The flesh raptors should appear around where the pupating zombie was, since they are coming out of the pupating zombie

# Screenshots

Before
Raptors
Corpse

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-5518-g9dbd68a [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda]
]
# Additional context
[CDDA save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/5147651/CDDA.save.zip)
Walk North, Pupating zombie is to the North East, near the corner of the collapsed tower. The zombie will die and you will be swarmed by raptors despite being far away.

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

SUMMARY: Balance "Adjust spawning and performance of flesh-raptors"

#### Purpose of change
Flesh raptors are appearing too early and are maybe a bit too tough.

Closes #43340, or should if I can get the spell code to cooperate.

#### Describe the solution

1. Fixes an error in the butcher shop json that had advanced fat zombie upgrades spawning on day 1
2. Increases mutation time on crawling zombies so raptors don't show up too soon
3. Makes raptors a little easier to hit
4. Attempts to make raptors spawn in the area of the bursting pupated zombie - this isn't quite working yet
5. Slightly increases cooldown timer on the raptor's deadly Impale attack.
6. Slow down regeneration on pupating zombie to match pupating zombie crawler
7. Slow down spellcasting speed on pupating zombies so they don't just insta-splode when they appear on screen.
8. Adds a "decoy" version of the pupating crawler and full size pupating zombie, that doesn't explode. This makes them harder to predict and also leaves room for them to evolve later.

#### Testing
Tested, all working good except they're not spawning around the pupating zombie corpse, they still appear around the player. Looking into that.

EDIT: @KorGgenT fixed it.
image

### CODE REVIEW

**Comment by Qrox** [comment]
I've also noticed that sometimes when a pupating zombie dies a message "failed to place monster" is displayed.

**Comment by I-am-Erk** [comment]
@korggent reminder to see if you can see the problem with the spell json

File: data/json/monster_special_attacks/spells.json [hunk 1]
```diff
@@ -118,14 +118,15 @@
...
-    "max_damage": 4,
-    "min_range": 3,
-    "max_range": 3,
-    "min_aoe": 2,
+    "valid_targets": [ "ground", "self" ],
+    "hit_self": true,
+    "min_damage": 1,
+    "max_damage": 2,
+    "min_range": 0,
+    "max_range": 1,
```
Comment (KorGgenT) [inline data/json/monster_special_attacks/spells.json:128]:
```suggestion
```
this spell has no level, so 0 is the max range this spell will be. 0 is the default, so no reason to bloat the json

File: data/json/monster_special_attacks/spells.json [hunk 2]
```diff
@@ -118,14 +118,15 @@
...
+    "min_aoe": 1,
     "max_aoe": 2,
```
Comment (KorGgenT) [inline data/json/monster_special_attacks/spells.json:130]:
```suggestion
    "min_aoe": 1,
    "max_aoe": 1,
```
max level 0 and no aoe_increment means these should be the same like for range

File: data/json/monster_special_attacks/spells.json [hunk 3]
```diff
@@ -118,14 +118,15 @@
...
+    "description": "Summons a few zombiespawn raptors.",
     "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT" ],
-    "valid_targets": [ "ground", "hostile" ],
-    "min_damage": 3,
-    "max_damage": 4,
-    "min_range": 3,
-    "max_range": 3,
-    "min_aoe": 2,
+    "valid_targets": [ "ground", "self" ],
+    "hit_self": true,
```
Comment (KorGgenT) [inline data/json/monster_special_attacks/spells.json:124]:
```suggestion
```
this needs to go in the monster attack, not here in the spell definition.

File: data/json/monster_special_attacks/spells.json [hunk 4]
```diff
@@ -138,13 +139,14 @@
...
     "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT" ],
-    "valid_targets": [ "ground", "hostile" ],
-    "min_damage": 6,
-    "max_damage": 8,
-    "min_range": 3,
-    "max_range": 3,
-    "min_aoe": 4,
-    "max_aoe": 5,
+    "valid_targets": [ "ground", "self" ],
+    "hit_self": true,
```
Comment (KorGgenT) [inline data/json/monster_special_attacks/spells.json:143]:
```suggestion
```
this needs to go in the monster attack, not here in the spell definition.

File: data/json/monster_special_attacks/spells.json [hunk 5]
```diff
@@ -138,13 +139,14 @@
...
+    "min_damage": 3,
+    "max_damage": 6,
+    "min_range": 0,
+    "max_range": 1,
```
Comment (KorGgenT) [inline data/json/monster_special_attacks/spells.json:147]:
```suggestion
```
this spell has no level, so 0 is the max range this spell will be. 0 is the default, so no reason to bloat the json

File: data/json/monster_special_attacks/spells.json [hunk 6]
```diff
@@ -138,13 +139,14 @@
...
+    "min_aoe": 2,
+    "max_aoe": 3,
```
Comment (KorGgenT) [inline data/json/monster_special_attacks/spells.json:149]:
```suggestion
    "min_aoe": 2,
    "max_aoe": 2,
```
max level 0 and no aoe_increment means these should be the same like for range

**Comment by KorGgenT** [comment]
not sure how to do an easy diff but like this
```

  {
    "id": "mon_zombie_crawler_pupa",
    "type": "MONSTER",
    "name": { "str": "pulsing zombie crawler" },
    "description": "This half of a human corpse is wrapped in sticky black fibers that cover everything neck down.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
    "copy-from": "mon_zombie_crawler",
    "armor_bash": 7,
    "armor_cut": 5,
    "armor_bullet": 5,
    "special_attacks": [
      { "type": "bite", "cooldown": 3 },
      {
        "type": "spell",
        "spell_data": { "id": "small_raptor_spawn", "hit_self": true },
        "cooldown": 20,
        "monster_message": "The pulsing zombie crawler bursts, and gore-smeared winged beasts climb out of the corpse!"
      }
    ],
    "regenerates": 10,
    "extend": { "flags": [ "SLUDGETRAIL", "SLUDGEPROOF" ] }
  },
```

**Comment by irwiss** [comment]
Can you also look into regen rate?

20 regen + bullet armor is like quarter of a shoggoth and twice a molerat,
makes low calibers essentially useless, the only way to kill it is emptying entire
magazines in bursts and not getting seen while you're at it

**Comment by I-am-Erk** [comment]
Thanks for spotting that @irwiss, looks like I forgot to turn it down on the full sized pupating zombie when I turned it down on the smaller one

**Pull request merged**